### DISCUSSION TITLE
Understanding Simulation error while simulating LAM degradation in negative electrode for LFP model

### DISCUSSION THREAD

**Discussion by ganmad** [pybamm-team/PyBaMM#2873] [Q&A]
Dear PyBaMM community,

I request your inputs on understanding and resolving Simulation Error that is arising while simulating LAM in negative electrode of LFP 18650 cell. 

The notebook on speeding up the solver [https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/solvers/speed-up-solver.ipynb](url) is quite useful and helped me to understand some of the solver options. However,  I still have further queries to understand my simulation error. 

Here is the exact error when model stops at:

"[ERROR] callbacks.on_experiment_error(226): Simulation error: Maximum number of decreased steps occurred at t=538000.4382690531. Try solving the model up to this time only or reducing dt_max (currently, dt_max=3.873579545457808) and/or reducing the size of the time steps or period of the experiment."

I would like to understand why the error is popping up even for finer mesh and higher relative tolerance. I am using CasadiSolver in safe mode. I have tried reducing dt_max as well. The model is DFN. 

What is the option to reduce time steps when we use pybamm.Exeperiment class? Is reducing dt_max not sufficient? 
I have not used any period condition to speed up the solver.  

Is there a better way to handle steep changes in OCVs at lower and higher SoCs? I am using Interpolant class for OCV csv data.

I am using the "pybamm.Experiment" class to simulate capacity fade due to LAM (stress-driven)  with particle mechanics (swelling only) during high current CC cycling (8C discharge, 1.25C charge).

The model works fine upto this time and predicts 8% capacity fade. I understand errors like "Jacobian failed" and "Linear solve failed" during time stepping when Experiments class is being used with events is harmless.

When I ran solution.termination, it shows 'event: Charge voltage cut-off [V] [experiment]' as the output. 

I checked for physical parameter such as SoCs of electrodes and electrolyte concentration, terminal voltage. They are within the range and I didn't find any abnormality at the failure. 

I have tried changing total number of cycles and reduced voltage cut-offs but no use. Interestingly, the model works find without any degradation mode and runs smoothly for any large number of cycles. 

Do you have any further suggestions to resolve this error? 

Thanks.

**Comment by RuiheLi** [comment]
Hi, I am interested in the final status of everything right before the failure point. I am wondering if we run the model extactly up to t=538000 seconds, and save the status, then we can:
(1) check everything
(2) do very small pertubation on the cell, by using `model.set_initial_conditions_from`, and do like `0.1C charge for 1 second (0.1 second period)` there may be a chance that we found something change dramatically, which quickly fail the simulation. 
I have this experience when trying to do GITT at low SOC, or high SOC, when the current is very high, the voltage changes quickly and exceeds the voltage cut-off, so even 1 second's silumation doesn't happen. That is why the core PyBaMM developer kindly add the function to skip the un-suitable steps and create an empty solution.
